src/CheckboxSelector/widget/checkboxselector.js

```diff
@@ -1,459 +1,418 @@
+				var thead = domConstruct.create('thead', { 'class': 'mendixDataGrid_gridHead' });
+				var headTr = domConstruct.create('tr', { 'class': 'mendixDataGrid_tableRow' }, thead);
+				var checkall = '';
+				if (this.hasSelectAll) {
+					checkall = domConstruct.create('input', { type: 'checkbox', 'class': 'checkbox_checkall' });
+					var checkallBool = (selectedObjs.length == objs.length);
+					checkall.checked = checkallBool;
+					if (this.isInactive) {
+						domAttr.set(checkall, "disabled", "disabled");
+					}
+				}
+
+				var checkboxHeadTh = domConstruct.create('th',
+					{ 'class': 'mendixDataGrid_tableHead mendixDataGrid_tableHeadFirst' });
+				var div = domConstruct.create('div', { 'class': 'mendixDataGrid_headContainer' }, checkboxHeadTh);
+				domConstruct.create('div', { 'class': 'mendixDataGrid_sortIcon' }, div);
+				var lastDiv = domConstruct.create('div', { 'class': 'mendixDataGrid_columnCaption' }, div);
+				lastDiv.append(checkall);
+
+				domStyle.set(checkboxHeadTh, 'width', this.boxWidth + this.boxWidthUnit);
+
+				headTr.appendChild(checkboxHeadTh);
+
+				for (var k = 0; k < this.splits.length; k++) {
+					var headTh = domConstruct.create('th', { 'class': 'mendixDataGrid_tableHead', 'width': this.splits[k].displayWidth + '%' });
+					var firstDiv = domConstruct.create('div', { 'class': 'mendixDataGrid_headContainer' }, headTh);
+					domConstruct.create('div', { 'class': 'mendixDataGrid_sortIcon' }, firstDiv);
+					domConstruct.create('div', { 'class': 'mendixDataGrid_columnCaption', 'innerHTML': this.splits[k].headers }, firstDiv);
+
+					if (k == this.splits.length - 1)
+						domClass.add(headTh, 'mendixDataGrid_tableHeadLast');
+
+					headTr.appendChild(headTh);
 				}
+				table.appendChild(thead);
 			}
-			
-			var checkboxHeadTh = mxui.dom.th(
-					{ 'class': 'mendixDataGrid_tableHead mendixDataGrid_tableHeadFirst' },
-					mxui.dom.div(
-						{ 'class': 'mendixDataGrid_headContainer' },
-						mxui.dom.div({ 'class': 'mendixDataGrid_sortIcon' }),
-						mxui.dom.div({ 'class': 'mendixDataGrid_columnCaption' }, 
-								checkall
-						)
-					)
-				);
-			
-			this.boxWidth && dojo.style( checkboxHeadTh, { width: this.boxWidth + this.boxWidthUnit });
-			
-			headTr.appendChild(checkboxHeadTh);
-			
-			for (var k = 0; k < this.splits.length; k++) {
-				var headTh = mxui.dom.th({ 'class' : 'mendixDataGrid_tableHead', 'width' : this.splits[k].displayWidth+'%'},
-								mxui.dom.div({ 'class' : 'mendixDataGrid_headContainer' },
-									mxui.dom.div( { 'class' : 'mendixDataGrid_sortIcon' }),
-									mxui.dom.div({ 'class' : 'mendixDataGrid_columnCaption' }, this.splits[k].headers)));
-				
-				if (k == this.splits.length-1)
-					mxui.dom.addClass(headTh, 'mendixDataGrid_tableHeadLast');
-				
-				headTr.appendChild(headTh);
+
+			var tbody = domConstruct.create('tbody', { 'class': 'mendixDataGrid_gridBody' });
+			table.appendChild(tbody);
+
+			this.hoverNode = null;
```

Comment:
What happend with `.liveConnect` ? Do not remove CODE! Comment it out and make a note... like TODO or reason why it is not used any longer (This case, function is replaced with dojo on, line xxx) 
liveconnect is still in the private API (mx7.6) not sure if the signature is still the same